### PR TITLE
Add Custom Code View setting panel

### LINKED ISSUE #1883: Make it simple to add custom CSS/JS for a View
Right now, users have to use the Customizer, a Code block, or add Custom Content widgets to embed CSS. That's silly. Even before we have full Styles extension, we should have the ability to add custom JS and CSS code to a View.

### PR DESCRIPTION
**Pull request by zackkatz**
We still need to:

- [x] Make sure doesn't conflict with PDF for GravityView
- [x] Add setting to print scripts/styles in header or footer 
- [x] Test with Firefox
- [x] Test with Chrome
- [x] Add filters where needed, including script position
- [x] Add CodeMirror support (see PDF for GravityView)
- [x] Discuss potential security issues

Resolves #1883

### CODE REVIEW

File: future/includes/class-gv-settings-view.php
Comment (mrcasual) [inline future/includes/class-gv-settings-view.php:1]: @malayladu, please check https://www.gitkraken.com/learn/git/best-practices/git-commit-message#using-imperative-verb-form for commit message tips.

File: future/includes/class-gv-settings-view.php [hunk 1]
```diff
@@ -609,6 +609,19 @@ public static function defaults( $detailed = false, $group = null ) {
 					'show_in_shortcode' => false,
 					'full_width'        => true,
 				),
+				'where_to_embed_js_css' => array(
+					'label'             => __( 'Where to embed above Javascript & CSS?', 'gk-gravityview' ),
```
Comment (mrcasual) [inline future/includes/class-gv-settings-view.php:613]: @malayladu, it's `JavaScript`.

File: future/includes/class-gv-settings-view.php
Comment (malayladu) [reply future/includes/class-gv-settings-view.php:1]: Thanks @mrcasual

File: future/includes/class-gv-settings-view.php (hunk 1 above)
Comment (malayladu) [reply future/includes/class-gv-settings-view.php]: Got it.

**Comment by malayladu** [comment]
@mrcasual As we have an option to embed JS in Header & Footer and CSS in Header only. I have modified the setting that belongs to JS only. 

So, instead of the Radio button, I have used Checkbox to embed JS in the Footer (considering, we are embedding JS in the header by default)

Embed-Custom-Javascript-Setting

**Comment by malayladu** [comment]
@zackkatz @mrcasual I have tested it with Gravity PDF and it seems working without any issues.

**Comment by zackkatz** [comment]
@malayladu The question isn't Gravity PDF, it's **PDF for GravityView**. I've added this to your account on the GravityKit site.

**Comment by malayladu** [comment]
> @malayladu The question isn't Gravity PDF, it's **PDF for GravityView**. I've added this to your account on the GravityKit site.

Ohh...My bad. I will test it with **PDF for GravityView**. Thanks for the clarification and the access.

**Comment by malayladu** [comment]
@zackkatz It's working with **PDF for GravityView** without any issues.

**Comment by malayladu** [comment]
@zackkatz Curious to know why embedding JS in the footer setting was removed.

**Comment by malayladu** [comment]
> @zackkatz Curious to know why embedding JS in the footer setting was removed.

Ok. I got the answer [here](https://github.com/GravityKit/GravityView/pull/1884/commits/eb2ee39ecf963a054fd4df7245f96a931f5f1ed7).

**Pull request merged**